python/ray/data/tests/test_binary.py

```diff
@@ -195,6 +198,64 @@ def skip_unpartitioned(kv_dict):
     )
 
 
+def _gen_chunked_binary(
+    dir_path: str, total_size: int, max_file_size: Optional[int] = None
+):
+    chunk_size = max_file_size or 256 * MiB
+    num_chunks = total_size // chunk_size
+    remainder = total_size % chunk_size
+
+    if max_file_size is not None and max_file_size < total_size:
+        for i in range(num_chunks):
+            filename = f"part_{i}.bin"
+            with open(f"{dir_path}/{filename}", "wb") as f:
+                f.write(b"a" * chunk_size)
+
+                print(f">>> Written file: {filename}")
+
+    else:
+        with open(f"{dir_path}/chunk.bin", "wb") as f:
+            for i in range(num_chunks):
+                f.write(b"a" * chunk_size)
+
+                print(f">>> Written chunk #{i}")
+
+            if remainder:
+                f.write(b"a" * remainder)
+
+    print(f">>> Wrote chunked dataset at: {dir_path}")
+
+
+@pytest.mark.parametrize(
+    "col_name",
+    [
+        "bytes",
+        # TODO fix numpy conversion
+        # "text",
+    ],
+)
+def test_single_row_lt_2gb(ray_start_regular_shared, col_name):
+    with TemporaryDirectory() as tmp_dir:
```

Comment:
Any reason not to just use the `tmp_path` fixture?

Reply:
Only reason is that i didn't know that such exists!